src/main/java/uk/ac/ebi/ena/webin/cli/service/SampleService.java

```diff
@@ -47,32 +60,85 @@
         }
     }
 
-    private static class SampleResponse {
-        public int taxId;
-        public String id;
-        public String organism;
-        public String bioSampleId;
-        public boolean canBeReferenced;
+    /**
+     * Depending on what the given sample ID looks like, sample information will be retrieved from either ENA or Biosamples.
+     * Also, if the sample is not retrieved from Biosamples then an attempt will be made to retrieve it from ENA.
+     */
+    public Sample getSample(String sampleId) {
+        Sample sample = null;
+
+        if (sampleId.toUpperCase().startsWith(BIOSAMPLES_ID_PREFIX)) {
+            sample = getBiosamplesSample(sampleId);

```

Comment:
According to Dipayan, Biosamples as of right now does not support SRA accessions but can do so in future for better Biosamples and ENA integration.

Reply:
Updated the implementation so that even when an SRA accession is given, the sample is looked up on Biosamples if it is found to have an assigned biosamples ID in ENA.